src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -224,5 +224,143 @@ internal static LambdaExpression GetParameterExpression(Type parameterType, Quer
                 ? (T?)value
                 : (T?)queryArgument.DefaultValue;
         }
+
+        /// <summary>
+        /// Configures query arguments and a field resolver for the specified <see cref="FieldType"/>, overwriting
+        /// any existing configuration within <see cref="FieldType.Arguments"/>, <see cref="FieldType.Resolver"/>
+        /// and <see cref="FieldType.StreamResolver"/>.
+        /// <br/><br/>
+        /// For fields and properties, no query arguments are added and the field resolver simply pulls the appropriate
+        /// member from <see cref="IResolveFieldContext.Source"/>.
+        /// <br/><br/>
+        /// For methods, method arguments are iterated and processed by
+        /// <see cref="AutoRegisteringObjectGraphType{TSourceType}.GetArgumentInformation{TParameterType}(FieldType, ParameterInfo)">GetArgumentInformation</see>, building
+        /// a list of query arguments and expressions as necessary. Then a field resolver is built around the method.
+        /// </summary>
+        internal static void BuildFieldTypeForOutput(
```

Comment:
This just consolidates code from `AutoRegisteringObjectGraphType` and `AutoRegisteringInterfaceGraphType`.  There are no code changes within this method.

Reply:
I see thought there is too much copy-pasted code in `AutoRegisteredInterfaceGraphType`. Over time, this will become a problem and we can forget that the code needs to be changed in two places. Now we are hostages of the situation with the existing `ObjectGraphType<TSourceType>` base class. I propose to add additional class or even make `AutoRegisteringHelper` non-static and compose it with `AutoRegisteredInterfaceGraphType/`AutoRegisteredObjectGraphType` delegating all real work to methods in `AutoRegisteringHelper`.

Reply:
Well, I certainly don't want to change the OOP design of the class, if that's what you mean.  I'll deduplicate as much code as possible within that constraint.

Reply:
I don't know if it's for the best, but I deduplicated every last possible line of code.  🤷‍♂️ 

As of now, I do not want to change the class design, such as doing anything like this:

```cs
AutoRegisteringInterfaceGraphType<THandler, TSource>
    where THelper : AutoRegisteringHelper
```

Doing so would add a whole layer of complexity to override something; and it is often enough to require one or a couple lines of code in one of the various methods.

Reply:
Unfortunately the deduplication changed almost all the code I'd already written, so now the changes are a bit more extensive.

Reply:
I mean add private field of `AutoRegisteringHelper` type add call its methods. No changes in public API surface.

Reply:
Well then we can refactor the code later.  I've already wrote this entire PR twice over.